src/geotiffwriter.js

```diff
@@ -252,11 +252,12 @@ const encodeImage = (values, width, height, metadata) => {
 
   const prfx = new Uint8Array(encodeIfds([ifd]));
 
-  const img = new Uint8Array(values);
+  // This should allow other kinds of typed arrays to be used. Uint8 is just a view of the real underlying data.
+  const img = new Uint8Array(values.buffer);
 
   const samplesPerPixel = ifd[277];
 
-  const data = new Uint8Array(numBytesInIfd + (width * height * samplesPerPixel));
+  const data = new Uint8Array(numBytesInIfd + (img.length * samplesPerPixel));
```

Comment:
the parameter passed into Uint8Array is the number of bytes to allocate for the GeoTIFF.  I believe that img.length & samplesPerPixel would pass in 1/4 of the bytes we need as Float32 takes up 4 bytes.  Perhaps, you could take the approach of adding in some if statements to check whether someone wants to write an 8-bit or Float32 GeoTIFF and using [Float32Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Float32Array)

Reply:
I think the typed arrays compensate for that. 
```
> new Uint8Array((new Uint8Array([1,2,3,4])).buffer).length
4
> new Uint8Array((new Float32Array([1,2,3,4])).buffer).length
16
```

Reply:
Yup. Great idea!